python/console/console_editor.py

```diff
@@ -55,6 +55,49 @@
 from qgis.utils import OverrideCursor
 
 
+def findMinimalDistanceIndex(source, target):
+    """ Find the source substring index that most closely matches the target string"""
+    index = min(len(source), len(target))
+    try:
+        from Levenshtein import distance as lev
+    except ImportError:
+        return index
```

Comment:
Rather than returning index, it now uses `difflib.SequenceMatcher` as a distance function. SequenceMatcher is always available but performs slightly worse than Levenshtein. No point in pushing a message imho